src/pybind/mgr/cephadm/module.py

```diff
@@ -2859,6 +2866,36 @@ def set_prometheus_access_info(self, user: str, password: str) -> str:
         self.set_store(PrometheusService.USER_CFG_KEY, user)
         self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return 'prometheus credentials updated correctly'
+    
+    @handle_orch_error
+    def set_prometheus_targets(self, url: str) -> str:
+        services = self.describe_prometheus_service_spec()
+        if url not in services.targets:
+            services.targets.append(url)
+        else:
+            raise cherrypy.HTTPError(400, 'Target {} already exists'.format(url))
```

Comment:
why are we raising a cherrypy exception from cephadm. this looks wrong. Probably cephadm has its own error handlers that we can use here?